src/spark_rapids_ml/knn.py

```diff
@@ -0,0 +1,344 @@
+#
+# Copyright (c) 2022, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from typing import Any, Callable, Dict, List, Optional, Tuple, Type, Union
+
+import cudf
+import numpy as np
+import pandas as pd
+from pyspark.ml.param.shared import HasInputCol, HasLabelCol, Param
+from pyspark.sql import Column, DataFrame
+from pyspark.sql.functions import col, lit
+from pyspark.sql.types import (
+    ArrayType,
+    DoubleType,
+    FloatType,
+    IntegerType,
+    Row,
+    StructField,
+    StructType,
+)
+
+from spark_rapids_ml.core import (
+    INIT_PARAMETERS_NAME,
+    CumlInputType,
+    CumlT,
+    _CumlEstimatorSupervised,
+    _CumlModel,
+    _set_pyspark_cuml_cls_param_attrs,
+    alias,
+)
+from spark_rapids_ml.params import _CumlClass, _CumlParams
+
+
+class NearestNeighborsClass(_CumlClass):
+    @classmethod
+    def _cuml_cls(cls) -> List[type]:
+        from cuml import NearestNeighbors as cumlNearestNeighbors
+
+        return [cumlNearestNeighbors]
+
+    @classmethod
+    def _param_mapping(cls) -> Dict[str, Optional[str]]:
+        return {}
+
+    @classmethod
+    def _param_excludes(cls) -> List[str]:
+        return [
+            "handle",
+            "algorithm",
+            "metric",
+            "p",
+            "algo_params",
+            "metric_expanded",
+            "metric_params",
+            "output_type",
+        ]
+
+
+class _NearestNeighborsCumlParams(_CumlParams, HasInputCol, HasLabelCol):
+    """
+    Shared Spark Params for NearestNeighbor and NearestNeighborModel.
+    """
+
+    def setInputCol(self, value: str) -> "_NearestNeighborsCumlParams":
+        """
+        Sets the value of `inputCol`.
+        """
+        self.set_params(inputCol=value)
+        return self
+
```

Comment:
Do we also need `setLabelCol()`?

Reply:
Ah..., probably not. NN reused the labelCol to differentiate data and query vectors, but the standard NN application itself does not need the label column. I may be wrong.